### PR TITLE
Fix mAP for absent classes and add some nice explanation text.

### PR DESCRIPTION
**Pull request by vcarpani**
@hgaiser

### CODE REVIEW

File: keras_maskrcnn/bin/evaluate.py [hunk 1]
```diff
@@ -137,9 +137,15 @@ def main(args=None):
...
 
         # print evaluation
+        present_classes = 0
+        precision = 0
         for label, average_precision in average_precisions.items():
-            print(generator.label_to_name(label), '{:.4f}'.format(average_precision))
-        print('mAP: {:.4f}'.format(sum(average_precisions.values()) / len(average_precisions)))
+            print('{:.0f} instances of class'.format(average_precision[1]),
+                  generator.label_to_name(label), 'with average precision: {:.4f}'.format(average_precision[0]))
+            if(average_precision[1] > 0):
```
Comment (hgaiser) [inline keras_maskrcnn/bin/evaluate.py]: Could you change this to:
```python
if average_precision[1] > 0:
```

File: keras_maskrcnn/bin/evaluate.py [hunk 2]
```diff
@@ -137,9 +137,15 @@ def main(args=None):
         )
 
         # print evaluation
+        present_classes = 0
+        precision = 0
         for label, average_precision in average_precisions.items():
```
Comment (hgaiser) [inline keras_maskrcnn/bin/evaluate.py]: Maybe it's more verbose to change this to:
```python
for label, (average_precision, num_annotations) in average_precisions.items():
```

**Comment by vcarpani** [comment]
@hgaiser Done, it's needed only for the csv callback, not for the COCO one

File: keras_maskrcnn/bin/evaluate.py [hunk 3]
```diff
@@ -137,9 +137,15 @@ def main(args=None):
         )
 
         # print evaluation
-        for label, average_precision in average_precisions.items():
-            print(generator.label_to_name(label), '{:.4f}'.format(average_precision))
-        print('mAP: {:.4f}'.format(sum(average_precisions.values()) / len(average_precisions)))
+        present_classes = 0
+        precision = 0
+        for label, (average_precision, num_annotations ) in average_precisions.items():
```
Comment (enricoliscio) [inline keras_maskrcnn/bin/evaluate.py]: remove white space before ending parenthesis

File: keras_maskrcnn/callbacks/eval.py [hunk 4]
```diff
@@ -63,7 +63,16 @@ def on_epoch_end(self, epoch, logs={}):
             save_path=self.save_path
         )
 
-        self.mean_ap = sum(average_precisions.values()) / len(average_precisions)
+        present_classes = 0
+        precision = 0
+        for label, (average_precision, num_annotations ) in average_precisions.items():
```
Comment (enricoliscio) [inline keras_maskrcnn/callbacks/eval.py]: Again, parenthesis

**Comment by hgaiser** [comment]
I think when @enricoliscio comments are fixed it can be merged.

**Pull request merged**